src/uthash.h

```diff
@@ -999,6 +999,38 @@ for(((el)=(head)), ((tmp)=DECLTYPE(el)((head!=NULL)?(head)->hh.next:NULL));
   (el) != NULL; ((el)=(tmp)), ((tmp)=DECLTYPE(el)((tmp!=NULL)?(tmp)->hh.next:NULL)))
 #endif
 
+#ifdef NO_DECLTYPE
+#define HASH_TOP(hh,head,el,tmp,eof)                                             \
+do {                                                                             \
+  ((el)=(head));                                                                 \
+  ((*(char**)(&(tmp)))=(char*)((head!=NULL)?(head)->hh.next:NULL));              \
+  ((*eof)=((el) == NULL));                                                       \
+} while(0)
+#else
+#define HASH_TOP(hh,head,el,tmp,eof)                                             \
+do {                                                                             \
+  ((el)=(head));                                                                 \
+  ((tmp)=DECLTYPE(el)((head!=NULL)?(head)->hh.next:NULL));                       \
+  ((*eof)=((el) == NULL));                                                       \
+} while(0)
+#endif
+
+#ifdef NO_DECLTYPE
+#define HASH_NEXT(hh,head,el,tmp,eof)                                            \
+do {                                                                             \
+  ((el)=(tmp));                                                                  \
```

Comment:
Style nits: please remove the outer set of `()` here and similarly throughout; and add spaces around the operators. Also on lines 1022 and 1023 you're missing `()` around the macro parameters `tmp` and `eof`.

    (el) = (tmp);
    *(char**)&(tmp) = (char*)((tmp) ? (tmp)->hh.next : NULL);
    *(eof) = !(el);  // although you're getting rid of this one, see below